### PR TITLE
Add scalafix and fix most errors

### PR DESCRIPTION
**Pull request by Katrix**
Adds Scalafix, and fixes most warnings that it emits when running it. Won't update documentation just yet as there are still warnings left that I want to fix. Those will have to wait for another time though. 

I want to look around for some other stuff, and wait for some stuff before fixing those. Most notably I want the upload process changes in the bootstrap branch as I've already made quite a few changes in this area, and I don't want it to be too hard to merge in these changes.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ cache:
   directories:
     - $HOME/.ivy2/cache
     - $HOME/.sbt
+    - $HOME/.coursier/cache
```
Comment (felixoi) [inline .travis.yml:30]: We are not using coursier in this repo. Why we should cache it then?

Comment (Katrix) [reply .travis.yml:30]: We totally are. I just didn't know which directory to cache. Every time scalafmt runs it downloads stuff using coursier. Scalafix itself also uses coursier to download stuff.

**Comment by Katrix** [comment]
Fixed some bugs in the PR, in addition to #558 and maybe #568.

File: app/security/pgp/PGPPublicKeyInfo.scala [hunk 2]
```diff
@@ -37,93 +40,75 @@ object PGPPublicKeyInfo {
...
 
-          Logger.debug("Signatures:")
-          val sigIter = key.getSignatures
-          while (sigIter.hasNext) {
-            val sig: PGPSignature = sigIter.next().asInstanceOf[PGPSignature]
-            Logger.debug("\tCreated at: " + sig.getCreationTime)
-            Logger.debug("\tCertification: " + sig.isCertification)
-          }
+                Logger.debug("Users:")
+                val firstUser = key.getUserIDs.next()
```
Comment (felixoi) [inline app/security/pgp/PGPPublicKeyInfo.scala]: Throws an NullPointerException when adding a new key

**Review by felixoi — APPROVED** [review]

**Pull request merged**